CI failure: helm v2 `Object is of type 'unknown'`

`master` nightly builds started failing with the following errors:
```
Error: helm/v2/helm.ts(161,9): error TS2322: Type 'Output<Output<{ [key: string]: CustomResource; }> | undefined>' is not assignable to type 'Output<{ [key: string]: CustomResource; }>'.
  Type 'OutputInstance<Output<{ [key: string]: CustomResource; }> | undefined>' is not assignable to type 'Output<{ [key: string]: CustomResource; }>'.
    Type 'OutputInstance<Output<{ [key: string]: CustomResource; }> | undefined>' is not assignable to type 'OutputInstance<{ [key: string]: CustomResource; }>'.
      Type 'Output<{ [key: string]: CustomResource; }> | undefined' is not assignable to type '{ [key: string]: CustomResource; }'.
        Type 'undefined' is not assignable to type '{ [key: string]: CustomResource; }'.

Error: helm/v2/helm.ts(226,31): error TS2571: Object is of type 'unknown'.
Error: helm/v2/helm.ts(250,43): error TS2571: Object is of type 'unknown'.
```

Nothing changed in the k8s repo before the failures, so one of our dependencies must have changed.

--- a/provider/pkg/gen/nodejs-templates/helm/v2/helm.ts
+++ b/provider/pkg/gen/nodejs-templates/helm/v2/helm.ts
@@ -222,7 +222,7 @@ export class Chart extends yaml.CollectionComponentResource {
                             stdio: ['pipe', 'pipe', 'ignore'], // Suppress tiller version error
                         },
                     ).toString();
-                } catch (e) {
+                } catch (e: any) {
                     helmVer = e.stdout.toString();
                 }
 
@@ -245,7 +245,7 @@ export class Chart extends yaml.CollectionComponentResource {
                 ).toString();
                 return this.parseTemplate(
                     yamlStream, cfg.transformations, cfg.resourcePrefix, configDeps, cfg.namespace);
-            } catch (e) {
+            } catch (e: any) {
                 // Shed stack trace, only emit the error.
                 throw new pulumi.RunError(e.toString());
             } finally {
@@ -253,6 +253,9 @@ export class Chart extends yaml.CollectionComponentResource {
                 chartDir.removeCallback();
                 overrides.removeCallback();
             }
+
+            // Without this block, the TS compiler adds `| undefined` even though execution won't get here.
+            return pulumi.output<{[key: string]: pulumi.CustomResource}>({});
         });
 
         this.ready = this.resources.apply(m => Object.values(m));

--- a/sdk/nodejs/helm/v2/helm.ts
+++ b/sdk/nodejs/helm/v2/helm.ts
@@ -222,7 +222,7 @@ export class Chart extends yaml.CollectionComponentResource {
                             stdio: ['pipe', 'pipe', 'ignore'], // Suppress tiller version error
                         },
                     ).toString();
-                } catch (e) {
+                } catch (e: any) {
                     helmVer = e.stdout.toString();
                 }
 
@@ -245,7 +245,7 @@ export class Chart extends yaml.CollectionComponentResource {
                 ).toString();
                 return this.parseTemplate(
                     yamlStream, cfg.transformations, cfg.resourcePrefix, configDeps, cfg.namespace);
-            } catch (e) {
+            } catch (e: any) {
                 // Shed stack trace, only emit the error.
                 throw new pulumi.RunError(e.toString());
             } finally {
@@ -253,6 +253,9 @@ export class Chart extends yaml.CollectionComponentResource {
                 chartDir.removeCallback();
                 overrides.removeCallback();
             }
+
+            // Without this block, the TS compiler adds `| undefined` even though execution won't get here.
+            return pulumi.output<{[key: string]: pulumi.CustomResource}>({});
         });
 
         this.ready = this.resources.apply(m => Object.values(m));

